tests/plugins/test_pulp_tag.py

```diff
@@ -0,0 +1,262 @@
+"""
+Copyright (c) 2017 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+
+from __future__ import unicode_literals
+
+import os
+import sys
+
+from osbs.build.build_response import BuildResponse
+from atomic_reactor.core import DockerTasker
+from atomic_reactor.inner import DockerBuildWorkflow
+from atomic_reactor.plugin import PostBuildPluginsRunner, PluginFailedException
+from atomic_reactor.util import ImageName
+from atomic_reactor.build import BuildResult
+from atomic_reactor.plugins.build_orchestrate_build import (OrchestrateBuildPlugin,
+                                                            WORKSPACE_KEY_BUILD_INFO)
+from atomic_reactor.constants import PLUGIN_PULP_TAG_KEY
+
+try:
+    if sys.version_info.major > 2:
+        # importing dockpulp in Python 3 causes SyntaxError
+        raise ImportError
+
+    import dockpulp
+except (ImportError):
+    dockpulp = None
+
+import pytest
+from flexmock import flexmock
+from tests.constants import INPUT_IMAGE, SOURCE, MOCK
+if MOCK:
+    from tests.docker_mock import mock_docker
+
+
+class X(object):
+    image_id = INPUT_IMAGE
+    git_dockerfile_path = None
+    git_path = None
+    base_image = ImageName(repo="qwe", tag="asd")
+
+
+class BuildInfo(object):
+    def __init__(self, v1_image_id=None):
+        annotations = {'meta': 'test'}
+        if v1_image_id:
+            annotations['v1-image-id'] = v1_image_id
+
+        self.build = BuildResponse({'metadata': {'annotations': annotations}})
+
+
+def prepare(v1_image_ids={}):
+    if MOCK:
+        mock_docker()
+    tasker = DockerTasker()
+    workflow = DockerBuildWorkflow(SOURCE, "test-image")
+    setattr(workflow, 'builder', X())
+    setattr(workflow.builder, 'source', X())
+    setattr(workflow.builder.source, 'dockerfile_path', None)
+    setattr(workflow.builder.source, 'path', None)
+    setattr(workflow, 'tag_conf', X())
+    setattr(workflow.tag_conf, 'images', [ImageName(repo="image-name1"),
+                                          ImageName(namespace="prefix",
+                                                    repo="image-name2"),
+                                          ImageName(repo="image-name3", tag="asd")])
```

Comment:
We can set this to e.g.
```
setattr(workflow.tag_conf, 'images', [ImageName(namespace="prefix", repo="repository", tag="1.0"),
                                      ImageName(namespace="prefix", repo="repository", tag="1.0-1"),
                                      ImageName(namespace="prefix", repo="repository", tag="latest")])
```

to control what we expect updateRepo to be called with.